Further enhance creating the custom SSL certs (#2487)
theforeman/foreman-documentation: guides/common/assembly_configuring-satellite-custom-server-certificate.adoc

If you cannot use a {Project} self-signed certificate, you can configure {ProjectServer} to use an SSL certificate signed by an external certificate authority (CA).

When you configure {ProjectName} with custom SSL certificates, you must fulfill the following requirements:

* You must use the privacy-enhanced mail (PEM) encoding for the SSL certificates.
* You must not use the same SSL certificate for both {ProjectServer} and {SmartProxyServer}.
* The same CA must sign certificates for {ProjectServer} and {SmartProxyServer}.
* An SSL certificate must not also be a CA certificate.
* An SSL certificate must include a subject alt name (SAN) entry that matches the common name (CN).
* An SSL certificate must be allowed for Key Encipherment using a Key Usage extension.
* An SSL certificate must not have a shortname as the CN.
* You must not set a passphrase for the private key.

To configure your {ProjectServer} with a custom certificate, complete the following procedures:

. xref:creating-a-custom-ssl-certificate_{project-context}[]
. xref:Deploying_a_Custom_SSL_Certificate_to_Server_{project-context}[]
. xref:deploying-a-custom-ssl-certificate-to-hosts_{project-context}[]

> . If you have external {SmartProxyServers} registered to {ProjectServer}, you must configure them with custom SSL certificates.

Reviewer: ```suggestion
. If you have external {SmartProxyServers} registered to {ProjectServer}, configure them with custom SSL certificates.
```
Just to stick with instructional wording. (This is the last comment, I promise.)

Author: No worries :grinning:

= Configuring {ProjectServer} with a Custom SSL Certificate

By default, {ProjectName} uses a self-signed SSL certificate to enable encrypted communications between {ProjectServer}, external {SmartProxyServers}, and all hosts.
If you cannot use a {Project} self-signed certificate, you can configure {ProjectServer} to use an SSL certificate signed by an external certificate authority (CA).

When you configure {ProjectName} with custom SSL certificates, you must fulfill the following requirements:

> * You must use the Privacy-Enhanced Mail (PEM) encoding for the SSL certificates.

Reviewer 2: ```suggestion
* You must use the privacy-enhanced mail (PEM) encoding for the SSL certificates.
```

* You must not use the same SSL certificate for both {ProjectServer} and {SmartProxyServer}.
* The same CA must sign certificates for {ProjectServer} and {SmartProxyServer}.
* An SSL certificate must not also be a CA certificate.

> * An SSL certificate must include a Subject Alt Name (SAN) entry that matches the Common Name (CN).

Reviewer 2: ```suggestion
* An SSL certificate must include a subject alt name (SAN) entry that matches the common name (CN).
```

> * An SSL certificate must be allowed for Key Encipherment via a Key Usage extension.

Author: `Using` :+1:

* An SSL certificate must not have a shortname as the CN.
* You must not set a passphrase for the private key.

To configure your {ProjectServer} with a custom certificate, complete the following procedures:

. xref:creating-a-custom-ssl-certificate_{project-context}[]
. xref:Deploying_a_Custom_SSL_Certificate_to_Server_{project-context}[]
. xref:deploying-a-custom-ssl-certificate-to-hosts_{project-context}[]
. If you have external {SmartProxyServers} registered to {ProjectServer}, you must configure them with custom SSL certificates.

> The same certificate authority (CA) must sign certificates for {ProjectServer} and {SmartProxyServer}.

Reviewer: ```suggestion
The same CA must sign certificates for {ProjectServer} and {SmartProxyServer}.
```
Although I'm not sure why we're repeating this sentence when we already have it in the above requirements.

Author: Omitted :+1:

By default, {ProjectName} uses a self-signed SSL certificate to enable encrypted communications between {ProjectServer}, external {SmartProxyServers}, and all hosts.
If you cannot use a {Project} self-signed certificate, you can configure {ProjectServer} to use an SSL certificate signed by an external Certificate Authority.

> When you configure {ProductName} with custom certificates, you must follow the requirements mentioned below:

Reviewer: ```suggestion
When you configure {ProjectName} with custom certificates, you must fulfill the following requirements:
```

* You must use the Privacy-Enhanced Mail (PEM) encoding for the SSL certificates.

> * You cannot use the same certificate for both {ProjectServer} and {SmartProxyServer}.

Reviewer: ```suggestion
* You cannot use the same SSL certificate for both {ProjectServer} and {SmartProxyServer}.
```

* The same Certificate Authority must sign certificates for {ProjectServer} and {SmartProxyServer}.

> * Certificate cannot also be a CA certificate.

Reviewer: ```suggestion
* A SSL certificate cannot also be a CA certificate.
```
The same for the following list items.

Author: Used `An`

* Certificate must include a Subject Alt Name (SAN) entry that matches the Common Name (CN).
* Certificate must be allowed for Key Encipherment via a Key Usage extension.
* Certificate cannot have a shortname as the Common Name (CN).

> * A passphrase cannot be set for the private key.

Reviewer: I'd use "must not" instead of "cannot", because we want to forbid these things, even though they are technically possible.

@bangelic @asteflova Can you give your opinions on this, please?

Author: I agree to `must not` :+1:

> * A passphrase should not be set for the private key.

Reviewer 3: :+1: on [RFC2119](https://www.rfc-editor.org/rfc/rfc2119) use of must.
